### PR TITLE
Handle frame drop

### PR DESCRIPTION
**Pull request by yhao10**
Handle frame drop in fork kiss icp repo.

The only API change is in `kiss_icp.py`. Function changed from `def register_frame(self, frame, timestamps)` to `def register_frame(self, frame, timestamps, frame_delta=1)`

### CODE REVIEW

File: cpp/kiss_icp/core/Deskew.cpp [hunk 1]
```diff
@@ -37,8 +37,9 @@ namespace kiss_icp {
 std::vector<Eigen::Vector3d> DeSkewScan(const std::vector<Eigen::Vector3d> &frame,
                                         const std::vector<double> &timestamps,
                                         const Sophus::SE3d &start_pose,
-                                        const Sophus::SE3d &finish_pose) {
-    const auto delta_pose = (start_pose.inverse() * finish_pose).log();
+                                        const Sophus::SE3d &finish_pose,
+					int prev_frame_delta) {
```
Comment (Samahu) [inline cpp/kiss_icp/core/Deskew.cpp]: replace with spaces
```suggestion
                                        int prev_frame_delta) {
```

File: cpp/kiss_icp/core/Deskew.cpp [hunk 2]
```diff
@@ -37,8 +37,9 @@ namespace kiss_icp {
...
+    const auto delta_pose = (start_pose.inverse() * finish_pose).log() / prev_frame_delta;
```
Comment (Samahu) [inline cpp/kiss_icp/core/Deskew.cpp:45]: We may want to check if `prev_frame_delta` equals to zero before dividing

Comment (yhao10) [reply cpp/kiss_icp/core/Deskew.cpp:45]: addressed

**Review by Samahu — APPROVED** [review]

File: cpp/kiss_icp/core/Deskew.cpp [hunk 3]
```diff
@@ -37,8 +37,12 @@ namespace kiss_icp {
 std::vector<Eigen::Vector3d> DeSkewScan(const std::vector<Eigen::Vector3d> &frame,
                                         const std::vector<double> &timestamps,
                                         const Sophus::SE3d &start_pose,
-                                        const Sophus::SE3d &finish_pose) {
-    const auto delta_pose = (start_pose.inverse() * finish_pose).log();
+                                        const Sophus::SE3d &finish_pose,
+                                        int prev_frame_delta) {
+    if (prev_frame_delta <= 0) {
+	throw std::runtime_error("Math error. prev_frame_delta has to be greater than zero.\n");
```
Comment (Samahu) [inline cpp/kiss_icp/core/Deskew.cpp]: BTW, you don't need the "\n" at the end. This isn't a print statement.

File: python/kiss_icp/pybind/kiss_icp_pybind.cpp [hunk 4]
```diff
@@ -103,12 +103,12 @@ PYBIND11_MODULE(kiss_icp_pybind, m) {
     m.def(
         "_deskew_scan",
         [](const std::vector<Eigen::Vector3d> &frame, const std::vector<double> &timestamps,
-           const Eigen::Matrix4d &T_start, const Eigen::Matrix4d &T_finish) {
+           const Eigen::Matrix4d &T_start, const Eigen::Matrix4d &T_finish, int prev_frame_delta=1) {
```
Comment (Samahu) [inline python/kiss_icp/pybind/kiss_icp_pybind.cpp]: @yhao10 Did you test this? I don't think this is how you expose `prev_frame_delta` as an optional parameter to python using pybind.

Comment (Samahu) [reply python/kiss_icp/pybind/kiss_icp_pybind.cpp]: What I mean by that; can you invoke `self.compensator.deskew_scan(frame, self.poses, timestamps)` (i.e. without the `prev_frame_delta` parameter and the python invocation succeeded without complaining about the fourth argument?

Comment (yhao10) [reply python/kiss_icp/pybind/kiss_icp_pybind.cpp]: good catch. pybind has different default parameter format. fixed

**Pull request merged**